flocker/dockerplugin/test/test_api.py

```diff
@@ -438,10 +440,34 @@ def error():
             {u"Name": u"whatever"}, 423,
             {u"Err": "no good"})
 
+    def test_unsupported_method(self):
+        """
+        If an unsupported method is requested the appropriate response code is
+        returned.
+        """
+        return self.assertResponseCode(
+            b"BAD_METHOD", b"/VolumeDriver.Path", None, NOT_ALLOWED)
+
+    def test_unknown_uri(self):
+        """
+        If an unknown URI path is requested the appropriate response code is
+        returned.
+        """
+        return self.assertResponseCode(
+            b"BAD_METHOD", b"/xxxnotthere", None, NOT_FOUND)
+
+    def test_empty_host(self):
+        """
+        If an empty host header is sent to the Docker plugin it does not blow
+        up.
+        """
+        return self.assertResult(b"POST", b"/Plugin.Activate", 12345, OK,
+                                 {u"Implements": [u"VolumeDriver"]},
+                                 additional_headers={b"Host": [""]})
+
 
 def _build_app(test):
     test.initialize()
     return VolumePlugin(
         test.volume_plugin_reactor, test.flocker_client, test.NODE_A).app
-RealTestsAPI, MemoryTestsAPI = buildIntegrationTests(
-    APITestsMixin, "API", _build_app)
+RealTestsAPI = buildUNIXIntegrationTests(APITestsMixin, "API", _build_app)
```

Comment:
No support for running over a TCP socket, then?  If this is because you never use Docker plugins like that, cool.

Any particular reason not to maintain the in-memory version?  Because it's useless (ie, it has never found a problem more quickly or identified the cause more clearly than the real-socket kind)?